### PR TITLE
[PRM-543]: added new due tag for LSP & LPP

### PR DESCRIPTION
**Pull request by BackRowOla**
LSP:
Screenshot 2021-08-24 at 15 28 41
LPP:
Screenshot 2021-08-24 at 15 28 52

### CODE REVIEW

File: app/viewmodels/SummaryCardHelper.scala [hunk 1]
```diff
@@ -322,9 +324,10 @@ class SummaryCardHelper @Inject()(link: views.html.components.link) extends Impl
 
       (penaltyAppealStatus, periodSubmissionStatus, penaltyPointStatus) match {
         case (Some(AppealStatusEnum.Accepted | AppealStatusEnum.Accepted_By_Tribunal), _, _) => renderTag(messages("status.cancelled"))
-        case (_, Some(Submitted), Due) => renderTag(messages("status.due"), "penalty-due-tag")
+//        case (_, Some(Submitted), Due) => renderTag(messages("status.due"), "penalty-due-tag")
```
Comment (ruthdas03) [inline app/viewmodels/SummaryCardHelper.scala:350]: Can the commented lines removed, or do we want to keep it.

File: app/viewmodels/SummaryCardHelper.scala [hunk 2]
```diff
@@ -375,4 +378,23 @@ class SummaryCardHelper @Inject()(link: views.html.components.link) extends Impl
...
+      val outstandingAmountDue = financial.get.outstandingAmountDue
+      if (outstandingAmountDue < amountDue && outstandingAmountDue > 0) {
+        renderTag(messages("status.partialPayment.due", outstandingAmountDue), "penalty-due-tag")
+      } else {
+        renderTag(messages("status.due"), "penalty-due-tag")
+      }
+  }else {
+    renderTag(messages("status.due"), "penalty-due-tag")
+  }
+
```
Comment (ruthdas03) [inline app/viewmodels/SummaryCardHelper.scala:410]: Please remove the commented lines.

File: app/viewmodels/SummaryCardHelper.scala [hunk 3]
```diff
@@ -375,4 +378,23 @@ class SummaryCardHelper @Inject()(link: views.html.components.link) extends Impl
       }
     }
   }
+
+  def showDueOrPartiallyPaidDueTag(financial: Option[Financial])(implicit messages: Messages): Tag = if (financial.isDefined){
+      val amountDue = financial.get.amountDue
+      val outstandingAmountDue = financial.get.outstandingAmountDue
+      if (outstandingAmountDue < amountDue && outstandingAmountDue > 0) {
```
Comment (ruthdas03) [inline app/viewmodels/SummaryCardHelper.scala]: In case of outstandingAmountDue=0.0 and amountDue = 0.0 we'd get a DUE tag, how are we handling this case.

**Review by ruthdas03 — COMMENTED** [review]
Seems ok, excepting a 'DUE' tag case and clean-up

File: app/viewmodels/SummaryCardHelper.scala (hunk 1 above)
Comment (BackRowOla) [reply app/viewmodels/SummaryCardHelper.scala:350]: Can be removed, will clean this up.

File: app/viewmodels/SummaryCardHelper.scala (hunk 3 above)
Comment (BackRowOla) [reply app/viewmodels/SummaryCardHelper.scala]: The function is only called in cases where something is due. Please correct me if I'm wrong, but in the case that the user has 0 outstanding amount and/or 0 amount due, wouldn't that mean they have fully paid their penalty and/or they don't have anything to pay. Which is covered by a different case in tagStatus()

Comment (BackRowOla) [reply app/viewmodels/SummaryCardHelper.scala]: Discussed on dev hangout, and has been resolved 👍

File: app/viewmodels/SummaryCardHelper.scala [hunk 4]
```diff
@@ -403,6 +402,7 @@ class SummaryCardHelper @Inject()(link: views.html.components.link) extends Impl
   }
 
   def showDueOrPartiallyPaidDueTag(financial: Option[Financial])(implicit messages: Messages): Tag = financial match {
+    case Some(financial) if (financial.outstandingAmountDue == 0) => renderTag(messages("status.paid"))
     case Some(financial) if (financial.outstandingAmountDue < financial.amountDue && financial.outstandingAmountDue > 0) =>
```
Comment (ruthdas03) [inline app/viewmodels/SummaryCardHelper.scala:406]: Do we need to check '&& financial.outstandingAmountDue > 0' looks like we are doing the check in the first case statement  'if (financial.outstandingAmountDue == 0)' . Please correct me if I'm wrong.

**Review by ruthdas03 — COMMENTED** [review]
Just another query in the 'if' clause.

File: app/viewmodels/SummaryCardHelper.scala (hunk 4 above)
Comment (BackRowOla) [reply app/viewmodels/SummaryCardHelper.scala:406]: You are correct, the '&&  financial.outstandingAmountDue > 0' check isn't needed anymore. Will edit and push up the change.

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**